### PR TITLE
Users/shmittal/utility change

### PR DESCRIPTION
**Pull request by 20shivangi**

### CODE REVIEW

**Comment by kilasuit** [comment]
Please add a description - watchers of this repo do read the PR's and cannot easily tell what this is and more importantly the intent of the PR just from commit messages, file changes and the PR Title

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 [hunk 1]
```diff
@@ -77,12 +77,8 @@ try
 {
     # Initialize Azure.
     Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
```
Comment (niadak) [inline Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: if (!$serviceName) check can be avoided

File: Tasks/AzurePowerShellV4/Utility.ps1 [hunk 2]
```diff
@@ -98,10 +71,9 @@ function  Get-RollForwardVersion {
...
         if(![string]::IsNullOrEmpty($rollForwardAzurePSVersion)) {
-            $hostedAgentAzureRmModulePath = Get-SavedModulePath -azurePowerShellVersion $rollForwardAzurePSVersion
-            $hostedAgentAzureModulePath = Get-SavedModulePath -azurePowerShellVersion $rollForwardAzurePSVersion -Classic
+            $hostedAgentAzModulePath = Get-SavedModulePath -azurePowerShellVersion $rollForwardAzurePSVersion
         
-            if((Test-Path -Path $hostedAgentAzureRmModulePath) -eq $true -or (Test-Path -Path $hostedAgentAzureModulePath) -eq $true) {
+            if((Test-Path -Path $hostedAgentAzModulePath) -eq $true) {
                 Write-Warning (Get-VstsLocString -Key "OverrideAzurePowerShellVersion" -ArgumentList $azurePowerShellVersion, $rollForwardAzurePSVersion)
                 return $rollForwardAzurePSVersion;
             }
```
Comment (niadak) [inline Tasks/AzurePowerShellV4/Utility.ps1]: No need of Roll Forward table now

File: Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1 [hunk 3]
```diff
@@ -21,18 +22,26 @@ function Initialize-AzModule {
 
 function Import-AzModule {
     [CmdletBinding()]
-    param()
+    param(
+        [string] $azVersion)
```
Comment (niadak) [inline Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1]: fix indentation

File: Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1 [hunk 4]
```diff
@@ -21,18 +22,26 @@ function Initialize-AzModule {
...
+        }
+        else{
+            $module = Get-Module -Name $moduleName -ListAvailable | Where-Object {$_.Version -eq $azVersion} | Select-Object -First 1
+        }
+      
         if (!$module) {
             Write-Verbose "No module found with name: $moduleName"
-            throw (Get-VstsLocString -Key AZ_ModuleNotFound -ArgumentList "Any version", "Az.Profile")
+            throw (Get-VstsLocString -Key AZ_ModuleNotFound -ArgumentList $azVersion, "Az.Accounts")
         }
```
Comment (niadak) [inline Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1]: This file change can be checked in to master independently

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 (hunk 1 above)
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: It is done

File: Tasks/AzurePowerShellV4/Utility.ps1 (hunk 2 above)
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/Utility.ps1]: It is done

File: Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1 (hunk 3 above)
Comment (20shivangi) [reply Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1]: It is done

**Pull request merged**